File: Difficulty: Easy/Missing in Array/missing-in-array.java
class Solution {
    int missingNum(int arr[]) {
        // code here
        int n = arr.length;

        // Correct expected sum from 1 to n+1
        long expectedSum = (long)(n + 1) * (n + 2) / 2;

        long actualSum = 0;
        for (int i = 0; i < n; i++) {
            actualSum += arr[i];
        }

        long missing = expectedSum - actualSum;

        return (int) missing;
    }
}

File: Difficulty: Easy/Missing in Array/missing-in-array.cpp
class Solution {
  public:
    int missingNum(vector<int>& arr) {
        // code here
        long long n=size(arr);
        
        long long total=(n+1)*(n+2)/2;
        long sum=0;
        
        for(int i=0; i<n;i++){
            sum+= arr[i];
        }
        
        return total-sum;
        
    }
};